auditlog/models.py

```diff
@@ -208,6 +212,18 @@ def _get_pk_value(self, instance):
             pk = self._get_pk_value(pk)
         return pk
 
+    def _get_serialized_data_or_none(self, instance):
+        from .registry import auditlog
+
+        registry_opts = auditlog.get_model_fields(instance.__class__)
+        if not registry_opts["serialize_data"]:
+            return None
+
+        kwargs = registry_opts.get("serialize_kwargs", {})
+        kwargs.setdefault("fields", [x.name for x in instance._meta.fields])
+        data = json.loads(serializers.serialize("json", (instance,), **kwargs))
```

Comment:
We have the logic which masks the values of certain fields (see `mask_fields` setting), we should respect this setting in all our serialization because they may be used for privacy or legal concerns.

Reply:
Well... I may have went overboard on this one but I do like a good challenge. I can appreciate the need to mask values. I currently work in a regulated industry and we may need this feature too. I've added a `MaskedDictionary` object that will recursively search for any flat and nested keys as specified by the `mask_fields` kwarg. Nesting can be specified with key pathing using a double underscore convention. There's a good test case that shows this better than I can explain it in writing.

This might be excessive. Let me know if that is too much for the requirements here. My thought was that if there is any nested serialization for related fields this recursive approach would be necessary. I didn't end up finding a great solution for nested related serialization though. See my points on the relevant comment for that.

Reply:
Thanks @sum-rock for implementing `MaskedDictionary`. I think it would be better to keep the mask field implementation simple and just mask the field in the model not related models.

Reply:
@hramezani I've removed the `MaskedDictionary` and implemented something much simpler. Let me know if this looks acceptable. I also moved this work over to the manager. It seemed to be the right change considering the other adds around ``serialize_auditlog_fields_only``.